src/nnet3/nnet-simple-component.h

```diff
@@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
   Component *CollapseWithNext(const FixedAffineComponent &next) const;
```

Comment:
About the second part, if a component does not have the kUpdatable flag set, then probably it doesn't inherit from UpdatableComponent and so SetZero(), NumParameters(), and DotProduct() might not be available. 
However I can check that if a component does not have kUpdatable flag and is an instance of UpdatableComponent then these methods should do nothing. 
Also I believe these should in a separate function something like TestNnetComponentUpdatability. Is that right?